acvm-repo/acir/src/native_types/expression/operators.rs

```diff
@@ -230,4 +218,264 @@ mod tests {
+        let a = Expression::from_str("w5").unwrap();
+        let b = Expression::from_str("w2").unwrap();
+
+        let result = (&a * &b).unwrap();
+        // Should be w2*w5, not w5*w2
+        assert_eq!(result.to_string(), "w2*w5");
+
+        // Enforce commutativity
+        assert_eq!(&a * &b, &b * &a);
+    }
+
+    #[test]
+    fn mul_result_is_sorted() {
+        // Verify the witness ordering in mul_terms is correct
+        let a = Expression::from_str("w3 + w1").unwrap();
+        let b = Expression::from_str("w4 + w2").unwrap();
+
+        let result = (&a * &b).unwrap();
+        // Verify that each mul_term has properly ordered witnesses (smaller first)
+        for (_, wl, wr) in &result.mul_terms {
+            assert!(wl <= wr, "Witnesses in mul_terms should be ordered");
+        }
+    }
+
+    #[test]
+    fn neg_reference() {
+        // Test negation of a reference (uses clone + in-place negate)
+        let a = Expression::from_str("2*w1*w2 + 3*w1 + 5*w2 + 7").unwrap();
+        let result = -&a;
+
+        assert_eq!(result.to_string(), "-2*w1*w2 - 3*w1 - 5*w2 - 7");
+
+        // Original should be unchanged
+        assert_eq!(a.to_string(), "2*w1*w2 + 3*w1 + 5*w2 + 7");
+    }
+
+    #[test]
+    fn neg_owned() {
+        // Test negation of an owned expression (in-place, no clone)
+        let a = Expression::from_str("2*w1*w2 + 3*w1 + 5*w2 + 7").unwrap();
+        let result = -a;
+
+        assert_eq!(result.to_string(), "-2*w1*w2 - 3*w1 - 5*w2 - 7");
+    }
+
+    #[test]
+    fn neg_zero() {
+        // Negating zero should give zero
+        let zero: Expression<FieldElement> = Expression::zero();
+        let result = -&zero;
+
+        assert_eq!(result, Expression::zero());
+    }
+
+    #[test]
+    fn neg_constant() {
+        // Negating a constant expression
+        let a = Expression::from_field(FieldElement::from(42u128));
+        let result = -a;
+
+        assert_eq!(result.q_c, FieldElement::from(-42i128));
+        assert!(result.mul_terms.is_empty());
+        assert!(result.linear_combinations.is_empty());
+    }
+
+    #[test]
+    fn neg_linear_only() {
+        // Negating an expression with only linear terms
+        let a = Expression::from_str("3*w1 + 5*w2 + 7").unwrap();
+        let result = -a;
+
+        assert_eq!(result.to_string(), "-3*w1 - 5*w2 - 7");
+    }
+
+    #[test]
+    fn neg_mul_only() {
+        // Negating an expression with only multiplication terms
+        let a = Expression::from_str("2*w1*w2 + 4*w3*w4").unwrap();
+        let result = -a;
+
+        assert_eq!(result.to_string(), "-2*w1*w2 - 4*w3*w4");
+    }
+
+    #[test]
+    fn double_neg() {
+        // Double negation should give back the original
+        let a = Expression::from_str("2*w1*w2 + 3*w1 + 5").unwrap();
+        let result = -(-a.clone());
+
+        assert_eq!(result, a);
+    }
+
+    #[test]
+    fn neg_preserves_structure() {
```

Comment:
This feels implicitly tested by the other tests but I guess it doesn't hurt to keep as we are checking the expression fields directly.